src/app.ts

```diff
@@ -13,14 +27,44 @@ interface TemperatureSummary {
   average: number
 }
 
+// I sorting by time and after city
 export function processReadings(readings: TemperatureReading[]): void {
-  // add here your code
+  for (const data of readings) {
+    const indextime: number = data.time.getTime()
+    const city = data.city
+    const tempe = data.temperature
+    if (indextime in allDataTmp && city in allDataTmp[indextime])
+      allDataTmp[indextime][city]?.push(tempe)
+    else {
+      if (!(indextime in allDataTmp)) {
+        const tmp: Dictionary<string, Array<number>> = { [city]: [tempe] }
+        allDataTmp[indextime] = tmp
+      } else allDataTmp[indextime][city] = [tempe]
+    }
+  }
 }
 
 export function getTemperatureSummary(
   date: Date,
   city: string,
 ): TemperatureSummary | null {
   //add here your code
+
+  const datecity = allDataTmp[date.getTime()][city]
+  if (typeof datecity === 'object') {
+    const tam = datecity['length']
+
+    const result: TemperatureSummary = {
+      first: datecity[0],
+      last: datecity[tam - 1],
+      high: Math.max(...datecity),
+      low: Math.min(...datecity),
+      average: datecity.reduce((a, b) => a + b, 0) / tam,
```

Comment:
Yes, first the code see  more reading and sorting and by other approach I will need a object and I would be modifying it, because the data is just being saved and if I, starting to calculate probably I me confused. Also I can manage the  processReadings  in function  and compact and one for( I am searching https://codeburst.io/javascript-finding-minimum-and-maximum-values-in-an-array-of-objects-329c5c7e22a2 )

Reply:
it will take more execution time since all the "fors" would be done while the other approach would do the calculations when necessary, I also think that once executed, these data will not be lost but saved in a database or file to so do not repeat the process.